### PR TITLE
refactor/monorepo

### LINKED ISSUE #5: vscode extention
### Details
As a developer I want my doc comments to be highlighted and auto-completed so that it increases my productivity

Either an existing extention is updated or a new dedicated one is created with the following features:
- syntax highlighting
- auto-completion of the comments
- consistency checks with the code

_Note: This issue can be splitted if needed_ 

### Acceptance Criteria
- documentation comments are highlighted following the standard delivered in #4
- auto-completion works for doc comments
- in-line red squiggles with a tooltip is being displayed if comments do not match the code

### PR DESCRIPTION
**Pull request by internnos**
closes #5 

provide vscode extension (without publishing it into the marketplace)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,16 +18,15 @@ jobs:
         with:
           node-version: 16.x
       - name: installing yarn
-        run: npm install -g yarn
+        run: cd cli && npm install -g yarn
```
Comment (Bernardstanislas) [inline .github/workflows/ci.yml:21]: You should use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-setting-default-run-step-options-for-a-job to specify the current directory

File: vscode-extension/client/out/extension.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { ExtensionContext } from 'vscode';
```
Comment (Bernardstanislas) [inline vscode-extension/client/out/extension.d.ts:1]: Shouldn't this folder by git-ignored?

File: vscode-extension/scripts/e2e.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+export CODE_TESTS_PATH="$(pwd)/client/out/test"
+export CODE_TESTS_WORKSPACE="$(pwd)/client/testFixture"
+
+node "$(pwd)/client/out/test/runTest"
```
Comment (Bernardstanislas) [inline vscode-extension/scripts/e2e.sh:6]: Where is this file?

**Comment by Bernardstanislas** [comment]
I am not sure the PR title fully reflects what's in here :)
Moreover, I do not agree it closes the #29 since the goal of this issue is to have the extension published, which is not –yet– the case here

**Comment by internnos** [comment]
> I am not sure the PR title fully reflects what's in here :) Moreover, I do not agree it closes the #29 since the goal of this issue is to have the extension published, which is not –yet– the case here

oh yeah sorry I linked the wrong issue, should be different one, thanks for pointing out!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (internnos) [reply .github/workflows/ci.yml:21]: aah gotcha, i just know about this. on it

File: vscode-extension/client/out/extension.d.ts (hunk 2 above)
Comment (internnos) [reply vscode-extension/client/out/extension.d.ts:1]: yes, you're correct, would remove this and apply .gitignore!

File: vscode-extension/scripts/e2e.sh (hunk 3 above)
Comment (internnos) [reply vscode-extension/scripts/e2e.sh:6]: i think i should have removed this, this is unecessary

**Comment by internnos** [comment]
i have resolved the issues

please let me know if there are any suggestions!

@Bernardstanislas

**Review by Bernardstanislas — APPROVED** [review]

**Pull request merged**